benchmarks/c/miscellaneous/thread_id.c

```diff
@@ -0,0 +1,29 @@
+#include <pthread.h>
+#include <stdatomic.h>
+#include <assert.h>
+#include <dat3m.h>
+
+/*
+    The test shows that pthread_self and __VERIFIER_tid return the same value.
+    Furthermore, it shows that the id written by pthread_create(&t, ...) into "t" matches the
+    the id returned by pthread_self within the spawned child thread.
+    Expected result: PASS
+*/
+
+pthread_t childTid;
+
+void *thread1(void *arg)
+{
+    childTid = pthread_self();
+    return NULL;
+}
+
+int main()
+{
+    pthread_t t1;
+    pthread_create(&t1, NULL, thread1, NULL);
+    pthread_join(t1, NULL);
+
+    assert(childTid == t1);
+    assert(__VERIFIER_tid() == pthread_self());
```

Comment:
How does this equality hold if both functions have different return type?

Reply:
I don't exactly know why the compiler doesn't complain here. Maybe on my system `pthread_t` is compatible with `unsigned int`.
As long as it compiles, the equality holds because internally we treat `pthread_t` as an integer.

Reply:
I just checked and on Linux, the compiler does not complain..